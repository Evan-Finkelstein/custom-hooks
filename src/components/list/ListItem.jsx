import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ListItem = ({ name, image, id }) => (
    <Link to={`./breed/${id}`}> <figure >
        <img style={{
            height: '50%',
            width: '50%',
        }} src={image.url} />
        <figcaption>{name}</figcaption>
    </figure></Link>
);

ListItem.propTypes = {
    name: PropTypes.string,
    image: PropTypes.string,
    id: PropTypes.string,
};

export default ListItem;
