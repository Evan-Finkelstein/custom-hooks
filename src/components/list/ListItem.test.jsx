import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ListItem from './ListItem';

describe('ListItem component', () => {
  afterEach(() => cleanup());
  it('renders ListItem', () => {
    const { asFragment } = render(<ListItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
