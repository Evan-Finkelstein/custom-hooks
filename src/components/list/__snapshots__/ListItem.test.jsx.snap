// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders ListItem 1`] = `
<DocumentFragment>
  <a
    href="/breed/1"
  >
     
    <figure>
      <img
        src="https://cdn2.thedogapi.com/images/BJa4kxc4X.jpg"
        style="height: 50%; width: 50%;"
      />
      <figcaption>
        Affenpinscher
      </figcaption>
    </figure>
  </a>
</DocumentFragment>
`;
